commands/sql/sqlsync.drush.inc

```diff
@@ -262,7 +262,19 @@ function drush_sqlsync_sql_sync($source, $destination) {
   if ($do_rsync) {
     if (!drush_get_option('no-dump')) {
       // Cleanup if this command created the dump file.
-      $rsync_options['remove-source-files'] = TRUE;
```

Comment:
sql-sync adds this regardless or any rsync-version option, it doesn't really matter where it is set, it's simply not taken into account, this code is simply adding this option and passed along to drush_invoke_process() slightly below.

Then when you get to the rsync part of the command, because of:

```php
  $options = _drush_build_rsync_options($additional_options, $include_settings_is_default);

  // Get all of the args and options that appear after the command name.
  $original_args = drush_get_original_cli_args_and_options();
  foreach ($original_args as $original_option) {
    if ($original_option{0} == '-') {
      $options .= ' ' . $original_option;
    }
  }
```

This is adding --remove-source-files again to the rsync command which fails.

The only thing this patch does, is an attempts to emulate what happens on `_drush_build_rsync_options` in regard to this specific rsync setting.